### PR TITLE
Inventory Handling and Entity Handling supported version update

### PR DESCRIPTION
**Pull request by ReinforceZwei**
Inventory handling was basically done. (Player inventory only)
I haven't tested chest or other container yet because I can't figure out how to open them :cry: 

Also I have limited the supported version of entity handling because I don't have time to test all the version now.

### CODE REVIEW

**Comment by ReinforceZwei** [comment]
I have a question about packet processing. Do I have to completely read the whole packet?
For example, a packet contains VarInt, then byte, bool, lastly string. I don’t need the string at the end, can I just ignore it?

File: MinecraftClient/Protocol/Handlers/Protocol18.cs [hunk 1]
```diff
@@ -1230,5 +1340,42 @@ public bool SendUseItemPacket(int hand)
...
+            catch (System.IO.IOException) { return false; }
+            catch (ObjectDisposedException) { return false; }
+        }
+
+        public bool SendHeldItemChange(short slot)
+        {
+            try
+            {
+                List<byte> packet = new List<byte>();
+                // short to byte (?
```
Comment (ORelio) [inline MinecraftClient/Protocol/Handlers/Protocol18.cs]: Seems like the field should indeed be a short here:
https://wiki.vg/Protocol#Held_Item_Change_.28serverbound.29
Serverbound is packet send by client to the server (while clientbound uses a byte)
Would be handy to add `GetShort()` inside [DataTypes.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/Protocol/Handlers/DataTypes.cs), then use `GetShort(slot)` here.

File: MinecraftClient/Commands/ChangeSlot.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    slot = Convert.ToInt16(getArg(command));
+                }
+                catch (FormatException)
+                {
+                    return "Could not change slot: Not a Number";
+                }
+                if (slot >= 1 && slot <= 9)
+                {
+                    if (handler.ChangeSlot(slot-=1))
```
[the remark this answers is not in the document]
Comment (ReinforceZwei) [reply MinecraftClient/Commands/ChangeSlot.cs]: I tried using slot - 1 or slot + 1, but the compiler complains about couldn’t convert Int32 to short.
I tried different approach and lastly found this, maybe the easiest solution.

File: MinecraftClient/Commands/ChangeSlot.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                    {
+                        return "Changed to slot " + (slot+=1);
```
[the remark this answers is not in the document]
Comment (ReinforceZwei) [reply MinecraftClient/Commands/ChangeSlot.cs]: Same as above.

File: MinecraftClient/MinecraftClient.csproj [hunk 4]
```diff
@@ -286,6 +289,7 @@
     <Content Include="Protocol\Dns\Records\totla.txt" />
     <Content Include="Resources\AppIcon.ico" />
   </ItemGroup>
+  <ItemGroup />
```
[the remark this answers is not in the document]
Comment (ReinforceZwei) [reply MinecraftClient/MinecraftClient.csproj]: I have no idea why those line are added :thinking:

File: MinecraftClient/Commands/ChangeSlot.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ORelio) [reply MinecraftClient/Commands/ChangeSlot.cs]: OK, lets keep this.

**Comment by ORelio** [comment]
Perfect. Also, replying to your question: You do not need to read fields left at the end of a packet because the whole packet is read anyway at the network level (it has a "packet length" field).

**Comment by ORelio** [comment]
(Note that it is true only for MC 1.7 owards. MC 1.6 and below have no packet lenght and all fields must be read, hence all the useless read operations in Protocol16.cs)

**Comment by ORelio** [comment]
Just pushed code documentation for the new features in 195e162c7db2d29ea1859cf10f323366fdcfe2d4
Changed a few method names, hope you were not already working on further additions 🙏

**Comment by ReinforceZwei** [comment]
Yes, I did not add any new code. I would like to focus on bug fixing now.
Thanks for the documentation by the way 😊

**Comment by ORelio** [comment]
Further documentation in 6929ae236a3f2e5e977b4b4ff92aaa5a42a914a4
Also renamed /getinventory to /inventory 😉

**Pull request merged**